### PR TITLE
Now able to deserialize tx with None "to address"

### PR DESCRIPTION
**Pull request by nebolax**
- Improves transaction deserialization edge cases handling
- Adds a test for this edge case

### CODE REVIEW

File: rotkehlchen/serialization/deserialize.py [hunk 1]
```diff
@@ -543,7 +543,7 @@ def deserialize_ethereum_transaction(
             timestamp = deserialize_timestamp(data['timeStamp'])
 
         from_address = deserialize_ethereum_address(data['from'])
-        to_address = deserialize_ethereum_address(data['to']) if data['to'] != '' else None
+        to_address = deserialize_ethereum_address(data['to']) if (data['to'] != '' and data['to'] is not None) else None  # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/serialization/deserialize.py:546]: This is way too big a condition for an one liner at this point. Please break it up.

File: rotkehlchen/tests/unit/test_serialization.py [hunk 2]
```diff
@@ -97,3 +108,40 @@ def test_deserialize_int_from_hex_or_int():
     assert deserialize_int_from_hex_or_int('0x1', 'whatever') == 1
     assert deserialize_int_from_hex_or_int('0x33', 'whatever') == 51
     assert deserialize_int_from_hex_or_int(66, 'whatever') == 66
+
+
+def test_deserialize_ethereum_transaction():
+    # Sometimes we can get transactions where "to address" is None
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/test_serialization.py:114]: This can happen yes. An ethereum transaction with a `to` missing means that the `from` address is deploying a new contract.

I would suggest that in this test test you test this kind of transaction. So rename it to `test_deserialize_deployment_ethereum_transaction` and remove the comments, or add it as a docstring.

In the ethereum world it's well known that empty `to` should mean contract deployment.

File: rotkehlchen/serialization/deserialize.py (hunk 1 above)
Comment (nebolax) [reply rotkehlchen/serialization/deserialize.py:546]: Sure!

**Review by LefterisJP — APPROVED** [review]
LGTM. Thanks Alexey
